pkg/probes/curl/curl_json.go

```diff
@@ -100,9 +99,26 @@ func (clp Probe) GetExpandedUserData(userDataVariables map[string]string) (strin
 	// Extract required variables specified in template (if any)
 	directivelessUserDataTemplate, requiredVariables := helpers.ExtractRequiredVariablesDirective(userDataTemplate)
 
+	curlCommand := curlgen.Options{
+		CaPath:          "/etc/pki/tls/certs/",
+		ProxyCaPath:     "/etc/pki/tls/certs/",
+		Retry:           3,
+		MaxTime:         userDataVariables["TIMEOUT"],
```

Comment:
You can't just move this up, the data for `TIMEOUT` for example is normalized via a function call on line 112/128.

Reply:
Done with reorg logic. Normalize first then command generation then validation of required vars.